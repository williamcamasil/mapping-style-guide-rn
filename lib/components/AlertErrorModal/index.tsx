import React from 'react';

import { withTheme } from '../../theme';
import AlertModal, { AlertModalPropsType } from '../AlertModal';

export type AlertErrorModalPropsType = AlertModalPropsType & {
  errorCode?: string;
};

const AlertErrorModal: React.FC<AlertErrorModalPropsType> = ({
  errorCode,
  ...others
}) => (
  <AlertModal
    {...others}
    extraText={errorCode ? `CÃ³digo: ${errorCode}` : undefined}
  />
);

export default withTheme(AlertErrorModal);
