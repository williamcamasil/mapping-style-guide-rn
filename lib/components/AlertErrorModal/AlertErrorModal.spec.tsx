import React from 'react';

import { render } from '@testing-library/react-native';

import AlertErrorModal from '.';

describe('AlertErrorModal snapshot', () => {
  it('Should render with extraText error', () => {
    const tree = render(
      <AlertErrorModal
        isFocused
        visible
        loading
        title="Title"
        description="Lorem ipsum is placeholder text commonly used in the graphic"
        errorCode="j23g4j23h4g"
        primaryButtonName="Primary Action"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render without extraText error', () => {
    const tree = render(
      <AlertErrorModal
        isFocused
        visible
        loading
        title="Title"
        description="Lorem ipsum is placeholder text commonly used in the graphic"
        primaryButtonName="Primary Action"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

