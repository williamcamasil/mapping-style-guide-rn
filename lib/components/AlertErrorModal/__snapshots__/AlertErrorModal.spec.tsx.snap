// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertErrorModal snapshot Should render with extraText error 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#3E4149",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 23,
                      "fontWeight": "bold",
                      "letterSpacing": -0.02,
                      "lineHeight": 28,
                      "marginBottom": 12,
                      "textAlign": "center",
                    }
                  }
                >
                  Title
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#676D79",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "letterSpacing": -0.02,
                      "lineHeight": 22,
                      "textAlign": "center",
                    }
                  }
                >
                  Lorem ipsum is placeholder text commonly used in the graphic
                </Text>
                <View>
                  <Text
                    style={
                      {
                        "height": 28,
                        "width": 28,
                      }
                    }
                  />
                  <Text
                    allowFontScaling={false}
                    style={
                      {
                        "color": "#676D79",
                        "flexShrink": 1,
                        "fontFamily": "Averta CY",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "letterSpacing": -0.02,
                        "lineHeight": 22,
                        "textAlign": "center",
                      }
                    }
                  >
                    CÃ³digo: j23g4j23h4g
                  </Text>
                </View>
                <View
                  style={
                    {
                      "marginTop": 40,
                    }
                  }
                >
                  <View
                    style={{}}
                    testID="container-button"
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": true,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#E95814",
                          "borderRadius": 12,
                          "borderWidth": 0,
                          "display": "flex",
                          "height": 64,
                          "justifyContent": "center",
                          "marginTop": 0,
                          "minWidth": 64,
                          "opacity": 1,
                          "paddingHorizontal": 32,
                        }
                      }
                      testID="alert-modal-primary-button"
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 1,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 0,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 0.5,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;

exports[`AlertErrorModal snapshot Should render without extraText error 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <RCTSafeAreaView
          pointerEvents="box-none"
          style={
            {
              "alignItems": "center",
              "bottom": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderRadius": 24,
                "margin": 8,
                "maxWidth": 340,
                "width": "100%",
              }
            }
          >
            <RCTScrollView
              bounces={false}
              contentContainerStyle={
                {
                  "padding": 28,
                }
              }
            >
              <View>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#3E4149",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 23,
                      "fontWeight": "bold",
                      "letterSpacing": -0.02,
                      "lineHeight": 28,
                      "marginBottom": 12,
                      "textAlign": "center",
                    }
                  }
                >
                  Title
                </Text>
                <Text
                  allowFontScaling={false}
                  style={
                    {
                      "color": "#676D79",
                      "flexShrink": 1,
                      "fontFamily": "Averta CY",
                      "fontSize": 16,
                      "fontWeight": "normal",
                      "letterSpacing": -0.02,
                      "lineHeight": 22,
                      "textAlign": "center",
                    }
                  }
                >
                  Lorem ipsum is placeholder text commonly used in the graphic
                </Text>
                <View
                  style={
                    {
                      "marginTop": 40,
                    }
                  }
                >
                  <View
                    style={{}}
                    testID="container-button"
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": true,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "backgroundColor": "#E95814",
                          "borderRadius": 12,
                          "borderWidth": 0,
                          "display": "flex",
                          "height": 64,
                          "justifyContent": "center",
                          "marginTop": 0,
                          "minWidth": 64,
                          "opacity": 1,
                          "paddingHorizontal": 32,
                        }
                      }
                      testID="alert-modal-primary-button"
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 1,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 0,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                        <View
                          style={
                            {
                              "borderColor": "hsla(0, 0%, 100%, 0.6)",
                              "borderRadius": 4,
                              "borderStyle": "solid",
                              "borderWidth": 0.5,
                              "height": 8,
                              "marginHorizontal": 4,
                              "overflow": "hidden",
                              "width": 8,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              {
                                "backgroundColor": "#FFFFFF",
                                "borderRadius": 4,
                                "height": "100%",
                                "opacity": 0.5,
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </RCTSafeAreaView>
      </View>
    </View>
  </View>
</View>
`;
